### PR TITLE
Adds ttl to expire caches.

### PR DESCRIPTION
**Pull request by ping-douglasl**
Adds optional TTL parameter to set method.
Adds documentation of the TTL functionality.
Adds tests to validate TTL functionality.

### CODE REVIEW

**Comment by ping-douglasl** [comment]
@philcockfield Would you mind taking a look at this implementation - I really liked your library - but I needed expiration in order to really use it. If you agree that it meets your quality requirements then I can import your library directly which would be great!

**Comment by philcockfield** [comment]
@ping-douglasl ... starting review shortly...
Thanks!

File: src/test/cache.expires.TEST.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  this.timeout(3000);
+  it('cache doesnt expire (various types)', (done) => {
+    const cache1 = new FileSystemCache({ basePath, ttl: 10 });
+    const cache2 = new FileSystemCache({ basePath, ttl: 10 });
+    
+    cache1.set('foo-1', 'bar-1');
+    cache1.set('foo-2', 'bar-2', 0);
+    cache1.set('foo-3', 'bar-3', 10)
+      .then(() => {
+        cache2.get('foo')
```
Comment (philcockfield) [inline src/test/cache.expires.TEST.ts:19]: nice!

**Review by philcockfield — APPROVED** [review]
LGTM. Nice one!  Beautiful style.

**Comment by philcockfield** [comment]
Published as `2.3.0`
https://www.npmjs.com/package/file-system-cache/v/2.3.0

Thanks @douglaslinsmeyer

**Comment by philcockfield** [comment]
resolves #27

**Comment by philcockfield** [comment]
closes #27

**Pull request merged**